src/community/ogcapi/ogcapi-features/src/test/java/org/geoserver/ogcapi/v1/features/CollectionsTest.java

```diff
@@ -67,6 +72,32 @@ public void testCollectionsJson() throws Exception {
         testCollectionsJson(json);
     }
 
+    @Test
+    public void testSkipMisconfigured() throws Exception {
+        // enable skipping of misconfigured layers
+        GeoServerInfo global = getGeoServer().getGlobal();
+        global.setResourceErrorHandling(ResourceErrorHandling.SKIP_MISCONFIGURED_LAYERS);
+        getGeoServer().save(global);
+        // manually misconfigure one layer
+        FeatureTypeInfo misconfigured =
+                getCatalog().getFeatureTypeByName(getLayerId(MockData.BUILDINGS));
+        misconfigured.setTitle(MISCONFIGURED_TITLE);
+        misconfigured.setAbstract(MISCONFIGURED_DESCRIPTION);
+        misconfigured.setOverridingServiceSRS(true);
+        misconfigured.getResponseSRS().addAll(Arrays.asList("3857", "32632"));
+        getCatalog().save(misconfigured);
+
+        DocumentContext json = getAsJSONPath("ogc/features/v1/collections", 200);
+        assertEquals(
+                "Misconfigured",
+                ((LinkedHashMap) ((JSONArray) json.read("collections")).get(16)).get("title"));
+
+        misconfigured.setLatLonBoundingBox(null);
+        getCatalog().save(misconfigured);
+        DocumentContext json2 = getAsJSONPath("ogc/features/v1/collections", 200);
+        assertNull(((JSONArray) json2.read("collections")).get(16));
```

Comment:
So what happens here... is the feature type being returned without the full configuration? It should really be skipped, not showing in the list (so I believe the iteration code should do a sort of loop, check the next, is it sane, if not move to the next, until one is sane or the iteration is completed).

Same goes for all services.

Reply:
Changed for all services